ouroboros-consensus/src/Ouroboros/Storage/FS/IO.hs

```diff
@@ -32,30 +34,36 @@ import qualified Ouroboros.Storage.Util.ErrorHandling as EH
 -- | File handlers for the IO instance for HasFS
 --
 -- We store the path the handle points to for better error messages
-data HandleIO = HandleIO {
-      handlePath :: FsPath
-    , handleIO   :: F.FHandle
-    }
-  deriving (Eq)
+type HandleIO = F.FHandle
+
+mkIOHasFS :: MountPoint -> IO (HasFS IO HandleIO)
+mkIOHasFS mount = do
+  initFid <- newMVar 0
+  return $ ioHasFS initFid mount
 
-ioHasFS :: MountPoint -> HasFS IO HandleIO
-ioHasFS mount = HasFS {
+ioHasFS :: MVar Int -> MountPoint -> HasFS IO HandleIO
```

Comment:
Now that we'll be using the `Eq` instance of handle `MVar`, we won't need this argument and the monadic `mkIOHasFS` function can go too. Most changes in the `StateMachine` module can be undone as well.